src/cortxfs/cortxfs_fops.c

```diff
@@ -103,27 +106,51 @@ int cfs_creat_ex(struct cfs_fs *cfs_fs, cfs_cred_t *cred, cfs_ino_t *parent,
 
 	RC_WRAP(kvs_begin_transaction, kvstor, &index);
 
-	RC_WRAP_LABEL(rc, out, cfs_creat, cfs_fs, cred, parent, name,
-		      mode, &object);
-	RC_WRAP_LABEL(rc, out, cfs_setattr, cfs_fs, cred, &object, stat_in,
+	RC_WRAP_LABEL(rc, out, cfs_creat, parent_fh, cred, name, mode,
+		      &object);
+	RC_WRAP_LABEL(rc, cleanup, cfs_setattr, cfs_fs, cred, &object, stat_in,
 		      stat_in_flags);
-	RC_WRAP_LABEL(rc, out, cfs_getattr, cfs_fs, cred, &object, stat_out);
+	RC_WRAP_LABEL(rc, cleanup, cfs_getattr, cfs_fs, cred, &object,
+		      stat_out);
 
 	RC_WRAP(kvs_end_transaction, kvstor, &index);
 
 	*newfile = object;
 	object = 0;
 
-out:
+cleanup:
 	if (object != 0) {
```

Comment:
Question: Do we need this condition? If control reaches to cleanup label, object is assumed to be non zero always?

Reply:
We need to do cleanup in error condition.